py/picca/delta_extraction/expected_fluxes/dr16_expected_flux.py

```diff
@@ -634,14 +640,17 @@ def compute_delta_stack(self, forests, stack_from_deltas=False):
                 bins = ((forest.log_lambda - Forest.log_lambda_min) /
                         Forest.delta_log_lambda + 0.5).astype(int)
             elif Forest.wave_solution == "lin":
-                bins = ((forest.lambda_ - Forest.log_lambda_min) /
-                        Forest.delta_log_lambda + 0.5).astype(int)
+                #TODO: this is definitely wrong, we do not have log-lambda quantities at this point, there might be subtleties as well
+
+                bins = ((forest.lambda_ - Forest.lambda_min) /
+                        Forest.delta_lambda + 0.5).astype(int)
             else:
                 raise ExpectedFluxError("Forest.wave_solution must be either "
                                         "'log' or 'linear'")
-            rebin = np.bincount(bins, weights=delta * weights)
+            #the indices are to exclude lambda>lambda_max from the stack                                        
```

Comment:
Why is this necessary? Forest only contains lambda<lambda_max, doen't it?

Reply:
It might actually be that one of the earlier steps led to having us pixels with higher lambda. I'll need to double check this again.